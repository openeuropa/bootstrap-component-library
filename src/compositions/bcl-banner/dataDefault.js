const drupalAttribute = require("drupal-attribute");

module.exports = {
  title: "EU Budget for the future",
  description: "Innovation, economy, environment and geopolitics",
  link: {
    label: "Subscribe",
    path: "/example.html",
    icon: {
      name: "chevron-right",
      path: "boostrap-icons.svg",
    },
    attributes: new drupalAttribute().addClass(["btn", "btn-primary"]),
  },
  attributes: new drupalAttribute().addClass([
    "bg-light",
    "text-dark",
    "text-center",
  ]),
};
