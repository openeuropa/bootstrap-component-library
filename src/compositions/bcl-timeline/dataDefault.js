const drupalAttribute = require("drupal-attribute");

module.exports = {
  id: 1,
  toggle_collapsed: "Show %d more items",
  toggle_expanded: "Hide %d items",
  hide: {
    from: 5,
    to: -2,
  },
  icons_path: "boostrap-icons.svg",
  items: [
    {
      label: "13 September 2017",
      title: "Item title",
      content:
        '<a href="/example">President Juncker\'s State of the Union speech</a>',
    },
    {
      label: "28-29 September 2017",
      content: '<a href="/example">Informal Digital Summit, Tallinn</a>',
    },
    {
      label: "14 November 2017",
      content:
        "Strengthening European identity through education and culture: European Commission's contribution to the Leaders' meeting.",
    },
    {
      label: "17 November 2017",
      content: '<a href="/example">Social Summit in Gothenburg, Sweden</a>',
    },
    {
      label: "6 December 2017",
      content:
        '<a href="/example">Economic and Monetary Union package of proposals</a>',
    },
    {
      label: "14-15 December 2017",
      content:
        '<a href="/example">EU Leaders\' meeting on migration, Brussels</a>',
    },
    {
      label: "15 December 2017",
      content: '<a href="/example">Euro Summit</a>',
    },
    {
      label: "6 February 2018",
      content: '<a href="/example">EU-Western Balkans Strategy</a>',
    },
    {
      label: "14 February 2018",
      content:
        '<a href="/example">Multiannual Financial Framework</a> and <a href="/example">institutional issues</a> - enhancing efficiency at the helm of the European Union',
    },
    {
      label: "23 February 2018",
      content: '<a href="/example">Informal European Council</a>',
    },
    {
      label: "22-23 March 2018",
      content: '<a href="/example">European Council</a>',
    },
    {
      label: "2 May 2018",
      content:
        '<a href="/example">Commission proposal: Long-term budget post-2020</a>',
    },
    {
      label: "15 May 2018",
      content:
        '<a href="/example">European Commissionâ€™s contribution to the Informal Leaders\' meeting</a>, Sofia, Bulgaria (16 May 2018)',
    },
    {
      label: "17 May 2018",
      content:
        '<a href="/example">EU-Western Balkans Summit</a> (Sofia, Bulgaria)',
    },
    {
      label: "23 May 2018",
      content:
        'Progress by Member States in meeting the conditions for adopting the euro - <a href="/example">reports</a> by the European Commission and ECB',
    },
    {
      label: "23 May 2018",
      content:
        '<a href="/example">European Semester</a>: European Commission publishes economic policy guidance for Member States',
    },
    {
      label: "29 May - 14 June 2018",
      content:
        '<a href="/example">EU budget for the future</a>: European Commission\'s legislative proposals for spending programmes',
    },
    {
      label: "28-29 June 2018",
      content:
        'Commission contribution to the European Council: <a href="/example">Economic and Monetary Union, migration, EU budget for the future</a>',
    },
    {
      label: "10 July 2018",
      content:
        'Ways to enhance subsidiarity, proportionality and better regulation in the daily operations of the EU: <a href="/example">Report</a> by the Task Force chaired by Commission First Vice-President Timmermans',
    },
    {
      label: "12 September 2018",
      content:
        '<a href="/example">President Juncker\'s State of the Union speech</a>',
    },
    {
      label: "20 September 2018",
      content:
        '<a href="/example">Informal European Council/Meeting of EU27 (Salzburg, Austria)</a>',
    },
    {
      label: "18 October 2018",
      content: '<a href="/example">European Council meeting</a>, Brussels',
    },
    {
      label: "21 November 2018",
      content:
        '<a href="/example">European Semester</a>: setting the economic and social priorities',
    },
    {
      label: "28 November 2018",
      content:
        'Commission contribution to the European Council on <a href="/example">Capital Markets Union and Banking Union</a>',
    },
    {
      label: "04 December 2018",
      content:
        'Commission contribution to the European Council on <a href="/example">long-term EU budget</a> and <a href="/example">migration</a>',
    },
    {
      label: "05 December 2018",
      content:
        'Commission contribution to the European Council on the <a href="/example">international role of the euro</a> and the <a href="/example">Action Plan on disinformation</a>',
    },
    {
      label: "11 December 2018",
      content:
        'Commission contribution to the European Council on the <a href="/example">Citizens\' Dialogues</a>',
    },
    {
      label: "11 December 2018",
      content:
        'Commission contribution to the <a href="/example">Euro Summit</a>',
    },
    {
      label: "13-14 December 2018",
      title: "Item title",
      content: "European Council and Euro Summit",
    },
  ],
  attributes: new drupalAttribute(),
};
