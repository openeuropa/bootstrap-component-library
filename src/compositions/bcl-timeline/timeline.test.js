import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import dataDefault from "@openeuropa/bcl-timeline/data";

const template = "@oe-bcl/bcl-timeline/timeline.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Timeline", () => {
  test("default renders correctly", () => {
    expect.assertions(1);
    return expect(render(dataDefault)).resolves.toMatchSnapshot();
  });

  test("default renders correctly with title and link", () => {
    expect.assertions(1);
    return expect(
      render({
        ...dataDefault,
        title: "Timeline test title",
        title_tag: "h6",
        title_link: {
          path: "/example.html",
          label: "Link label",
        },
      })
    ).resolves.toMatchSnapshot();
  });
});
