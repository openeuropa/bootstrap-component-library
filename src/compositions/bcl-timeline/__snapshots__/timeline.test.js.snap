// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Timeline default renders correctly 1`] = `
<jest>
  <ol
    class="bcl-timeline"
  >
    <li>
      <h5>
        13 September 2017
      </h5>
      <h6>
        Item title
      </h6>
      <div>
        <a
          href="/example"
        >
          President Juncker's State of the Union speech
        </a>
      </div>
    </li>
    <li>
      <h5>
        28-29 September 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          Informal Digital Summit, Tallinn
        </a>
      </div>
    </li>
    <li>
      <h5>
        14 November 2017
      </h5>
      <div>
        Strengthening European identity through education and culture: European Commission's contribution to the Leaders' meeting.
      </div>
    </li>
    <li>
      <h5>
        17 November 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          Social Summit in Gothenburg, Sweden
        </a>
      </div>
    </li>
    <li>
      <h5>
        6 December 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          Economic and Monetary Union package of proposals
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        14-15 December 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          EU Leaders' meeting on migration, Brussels
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        15 December 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          Euro Summit
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        6 February 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          EU-Western Balkans Strategy
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        14 February 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          Multiannual Financial Framework
        </a>
         and 
        <a
          href="/example"
        >
          institutional issues
        </a>
         - enhancing efficiency at the helm of the European Union
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        23 February 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          Informal European Council
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        22-23 March 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Council
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        2 May 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          Commission proposal: Long-term budget post-2020
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        15 May 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Commissionâ€™s contribution to the Informal Leaders' meeting
        </a>
        , Sofia, Bulgaria (16 May 2018)
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        17 May 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          EU-Western Balkans Summit
        </a>
         (Sofia, Bulgaria)
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        23 May 2018
      </h5>
      <div>
        Progress by Member States in meeting the conditions for adopting the euro - 
        <a
          href="/example"
        >
          reports
        </a>
         by the European Commission and ECB
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        23 May 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Semester
        </a>
        : European Commission publishes economic policy guidance for Member States
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        29 May - 14 June 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          EU budget for the future
        </a>
        : European Commission's legislative proposals for spending programmes
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        28-29 June 2018
      </h5>
      <div>
        Commission contribution to the European Council: 
        <a
          href="/example"
        >
          Economic and Monetary Union, migration, EU budget for the future
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        10 July 2018
      </h5>
      <div>
        Ways to enhance subsidiarity, proportionality and better regulation in the daily operations of the EU: 
        <a
          href="/example"
        >
          Report
        </a>
         by the Task Force chaired by Commission First Vice-President Timmermans
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        12 September 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          President Juncker's State of the Union speech
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        20 September 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          Informal European Council/Meeting of EU27 (Salzburg, Austria)
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        18 October 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Council meeting
        </a>
        , Brussels
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        21 November 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Semester
        </a>
        : setting the economic and social priorities
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        28 November 2018
      </h5>
      <div>
        Commission contribution to the European Council on 
        <a
          href="/example"
        >
          Capital Markets Union and Banking Union
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        04 December 2018
      </h5>
      <div>
        Commission contribution to the European Council on 
        <a
          href="/example"
        >
          long-term EU budget
        </a>
         and 
        <a
          href="/example"
        >
          migration
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        05 December 2018
      </h5>
      <div>
        Commission contribution to the European Council on the 
        <a
          href="/example"
        >
          international role of the euro
        </a>
         and the 
        <a
          href="/example"
        >
          Action Plan on disinformation
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        11 December 2018
      </h5>
      <div>
        Commission contribution to the European Council on the 
        <a
          href="/example"
        >
          Citizens' Dialogues
        </a>
      </div>
    </li>
    <li
      class="bcl-timeline__item--toggle"
    >
      <button
        aria-expanded="false"
        class="btn btn-light btn-md"
        data-bs-target=".collapse-1"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="label-collapsed"
        >
          Show 22 more items
        </span>
        <span
          class="label-expanded"
        >
          Hide 22 items
        </span>
        <svg
          class="ms-2-5 bi icon--fluid"
        >
          <use
            xlink:href="bootstrap-icons.svg#chevron-down"
          />
        </svg>
      </button>
    </li>
    <li>
      <h5>
        11 December 2018
      </h5>
      <div>
        Commission contribution to the Euro Summit
      </div>
    </li>
    <li>
      <h5>
        13-14 December 2018
      </h5>
      <h6>
        Item title
      </h6>
      <div>
        European Council and Euro Summit
      </div>
    </li>
  </ol>
</jest>
`;

exports[`OE - Timeline default renders correctly with title 1`] = `
<jest>
  <h6
    class="fw-bold mb-4 bcl-heading"
  >
    Timeline test title
  </h6>
  <ol
    class="bcl-timeline"
  >
    <li>
      <h5>
        13 September 2017
      </h5>
      <h6>
        Item title
      </h6>
      <div>
        <a
          href="/example"
        >
          President Juncker's State of the Union speech
        </a>
      </div>
    </li>
    <li>
      <h5>
        28-29 September 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          Informal Digital Summit, Tallinn
        </a>
      </div>
    </li>
    <li>
      <h5>
        14 November 2017
      </h5>
      <div>
        Strengthening European identity through education and culture: European Commission's contribution to the Leaders' meeting.
      </div>
    </li>
    <li>
      <h5>
        17 November 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          Social Summit in Gothenburg, Sweden
        </a>
      </div>
    </li>
    <li>
      <h5>
        6 December 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          Economic and Monetary Union package of proposals
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        14-15 December 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          EU Leaders' meeting on migration, Brussels
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        15 December 2017
      </h5>
      <div>
        <a
          href="/example"
        >
          Euro Summit
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        6 February 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          EU-Western Balkans Strategy
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        14 February 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          Multiannual Financial Framework
        </a>
         and 
        <a
          href="/example"
        >
          institutional issues
        </a>
         - enhancing efficiency at the helm of the European Union
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        23 February 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          Informal European Council
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        22-23 March 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Council
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        2 May 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          Commission proposal: Long-term budget post-2020
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        15 May 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Commissionâ€™s contribution to the Informal Leaders' meeting
        </a>
        , Sofia, Bulgaria (16 May 2018)
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        17 May 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          EU-Western Balkans Summit
        </a>
         (Sofia, Bulgaria)
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        23 May 2018
      </h5>
      <div>
        Progress by Member States in meeting the conditions for adopting the euro - 
        <a
          href="/example"
        >
          reports
        </a>
         by the European Commission and ECB
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        23 May 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Semester
        </a>
        : European Commission publishes economic policy guidance for Member States
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        29 May - 14 June 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          EU budget for the future
        </a>
        : European Commission's legislative proposals for spending programmes
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        28-29 June 2018
      </h5>
      <div>
        Commission contribution to the European Council: 
        <a
          href="/example"
        >
          Economic and Monetary Union, migration, EU budget for the future
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        10 July 2018
      </h5>
      <div>
        Ways to enhance subsidiarity, proportionality and better regulation in the daily operations of the EU: 
        <a
          href="/example"
        >
          Report
        </a>
         by the Task Force chaired by Commission First Vice-President Timmermans
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        12 September 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          President Juncker's State of the Union speech
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        20 September 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          Informal European Council/Meeting of EU27 (Salzburg, Austria)
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        18 October 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Council meeting
        </a>
        , Brussels
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        21 November 2018
      </h5>
      <div>
        <a
          href="/example"
        >
          European Semester
        </a>
        : setting the economic and social priorities
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        28 November 2018
      </h5>
      <div>
        Commission contribution to the European Council on 
        <a
          href="/example"
        >
          Capital Markets Union and Banking Union
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        04 December 2018
      </h5>
      <div>
        Commission contribution to the European Council on 
        <a
          href="/example"
        >
          long-term EU budget
        </a>
         and 
        <a
          href="/example"
        >
          migration
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        05 December 2018
      </h5>
      <div>
        Commission contribution to the European Council on the 
        <a
          href="/example"
        >
          international role of the euro
        </a>
         and the 
        <a
          href="/example"
        >
          Action Plan on disinformation
        </a>
      </div>
    </li>
    <li
      class="collapse collapse-1"
    >
      <h5>
        11 December 2018
      </h5>
      <div>
        Commission contribution to the European Council on the 
        <a
          href="/example"
        >
          Citizens' Dialogues
        </a>
      </div>
    </li>
    <li
      class="bcl-timeline__item--toggle"
    >
      <button
        aria-expanded="false"
        class="btn btn-light btn-md"
        data-bs-target=".collapse-1"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="label-collapsed"
        >
          Show 22 more items
        </span>
        <span
          class="label-expanded"
        >
          Hide 22 items
        </span>
        <svg
          class="ms-2-5 bi icon--fluid"
        >
          <use
            xlink:href="bootstrap-icons.svg#chevron-down"
          />
        </svg>
      </button>
    </li>
    <li>
      <h5>
        11 December 2018
      </h5>
      <div>
        Commission contribution to the Euro Summit
      </div>
    </li>
    <li>
      <h5>
        13-14 December 2018
      </h5>
      <h6>
        Item title
      </h6>
      <div>
        European Council and Euro Summit
      </div>
    </li>
  </ol>
</jest>
`;
